src/main/java/org/openrewrite/java/dropwizard/annotation/AddAnnotationIfSuperTypeExists.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright 2025 the original author or authors.
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * https://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.openrewrite.java.dropwizard.annotation;
+
+import org.openrewrite.java.tree.J;
+import org.openrewrite.java.tree.JavaType;
+import org.openrewrite.java.tree.TypeUtils;
+
+import java.util.Objects;
+
+public class AddAnnotationIfSuperTypeExists extends AddClassAnnotation {
+
+    private final String targetSupertypeNames;
+
+    public AddAnnotationIfSuperTypeExists(
+            String annotationToAdd, String targetSupertypeNames, boolean annotateInnerClasses) {
+        super(annotationToAdd, annotateInnerClasses);
+        this.targetSupertypeNames = targetSupertypeNames;
+    }
```

Comment:
We try to avoid inheritance with recipes; the `targetSupertypeNames` I'd expect to have an `@Option` annotation before 
I expect calling this from yaml recipes would work; perhaps good to use that, and extract the shared visitor to a utility class used by both regular Recipe classes.

Reply:
To verify this works it might make sense to already pull in some of the declarative recipes into this PR, if that's an option.